src/components/sidebar-navigation/SidebarNavigationAccordionItem.tsx

```diff
@@ -142,18 +142,19 @@ const AccordionItemPanel = styled.section`
   padding-right: ${cssVar('dimension-space-200')};
   border-left: ${cssVar('border-width-default')} solid ${cssVar('color-border-weak')};
 
-  [data-sidebar-collapsed='true'] & {
+  // The children SidebarNavigationItems rely on this css property to set their display value, falling back to flex if not inside an accordion
+  --sidebar-navigation-accordion-children-display: flex;
+
+  // We force the children SidebarNavigationItems to be hidden when the accordion is closed or the sidebar is closed
+  &[data-accordion-open='false'] {
+    --sidebar-navigation-accordion-children-display: none;
+  }
```

Comment:
The comment is a bit confusing. How about
`The next two rules hide the child SidebarNavigationItems when the accordion is closed, and also when the sidebar is neither docked nor open (hovered or focused).`